url75694980d5_fabioalmeid_tp01ftc_tgz-pJ8-agentes_jade_Centralizador_ExemploParserGramaticaCentralizadorJ8/src/agentes/jade/Paciente/AgentePaciente.java

```diff
@@ -0,0 +1,361 @@
+import gramatica.Centralizador.Absyn.EApply2;
+import gramatica.Centralizador.Absyn.ECollect1;
+import gramatica.Centralizador.Absyn.EData1;
+import gramatica.Centralizador.Absyn.EData2;
+import gramatica.Centralizador.Absyn.EData3;
+import gramatica.Centralizador.Absyn.EData4;
+import gramatica.Centralizador.Absyn.ERemedy1;
+import gramatica.Centralizador.Absyn.ERemedy2;
+import gramatica.Monitor.Absyn.EDados;
+import gramatica.Monitor.Absyn.EDados1;
+import gramatica.Monitor.Absyn.EDados2;
+import gramatica.Monitor.Absyn.EDados3;
+import gramatica.Monitor.Absyn.EQuantidade;
+import jade.core.AID;
+import jade.core.Agent;
+import jade.core.behaviours.CyclicBehaviour;
+import jade.core.behaviours.TickerBehaviour;
+import jade.domain.DFService;
+import jade.domain.FIPAException;
+import jade.domain.FIPAAgentManagement.DFAgentDescription;
+import jade.domain.FIPAAgentManagement.ServiceDescription;
+import jade.lang.acl.ACLMessage;
+import jade.lang.acl.MessageTemplate;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import util.UpdateAgentList;
+
+import agentes.jade.Atuador.TarefaAtuador;
+import agentes.jade.Centralizador.GrammarParserCentralizador;
+import agentes.jade.Centralizador.Medicamento;
+import agentes.jade.Centralizador.TarefaCentralizador;
+import agentes.jade.Monitor.Afericao;
+import agentes.jade.Monitor.TarefaMonitor;
+
+public class AgentePaciente extends Agent {
+	private static final long INTERVALO_AGENTES = 5000; // 5s
+	private static final long INTERVALO_ATUALIZACAO = 6000; // 10s
+//	private static final int MAX_TICKS_UNCHANGED = 5;
+	private static final int MAX_TICKS_UNCHANGED = 2;
+	private List<AID> monitor;
+	private List<AID> atuador;
+	
+	protected void setup() {
+		// Regista o paciente no servico de paginas amarelas
+		DFAgentDescription dfd = new DFAgentDescription();
+		dfd.setName(getAID());
+		ServiceDescription sd = new ServiceDescription();
+		sd.setType("paciente");
+		sd.setName(getName());
+		dfd.addServices(sd);
+		try {
+			DFService.register(this, dfd);
+		} catch (FIPAException fe) {
+			fe.printStackTrace();
+		}
+		
+		// atualiza lista de monitores e atuadores
+		addBehaviour(new UpdateAgentsTempBehaviour(this, INTERVALO_AGENTES));
+		
+		// ouve requisicoes dos atuadores e monitores
+		addBehaviour(new ListenBehaviour());
+		
+		// adiciona comportamento de mudanca de temperatura
+		addBehaviour(new UpdateTemperatureBehaviour(this, INTERVALO_ATUALIZACAO));
+		
+		// adiciona comportamento de mudanca de hemoglobina		
+		addBehaviour(new UpdateHemoglobinaBehaviour(this, INTERVALO_ATUALIZACAO));
+		
+		// adiciona comportamento de mudanca de bilirrubina
+		addBehaviour(new UpdateBilirrubinaBehaviour(this, INTERVALO_ATUALIZACAO));
+
+		// adiciona comportamento de mudanca de pressao		
+		addBehaviour(new UpdatePressaoBehaviour(this, INTERVALO_ATUALIZACAO));
+		
+	}
+	
+	class ListenBehaviour extends CyclicBehaviour {
+
+		@Override
+		public void action() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 85 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.